### ISSUE TITLE
Underscore not being installed, as part of steps in README?

### ISSUE THREAD

**Issue by victorhooi** [hubot-scripts/hubot-group-alias#9]
This might be my n00bness with npm/node.

However, I followed the instructions in the README - namely running `npm i --save hubot-group-alias `, and adding `hubot-group-alias` to `external-scripts.json`.

However, when I try to startup Hubot, I get:
```
bash-3.2$ bin/hubot
connect deprecated multipart: use parser (multiparty, busboy, formidable) npm module instead node_modules/hubot/node_modules/express/node_modules/connect/lib/middleware/bodyParser.js:56:20
connect deprecated limit: Restrict request size at location of read node_modules/hubot/node_modules/express/node_modules/connect/lib/middleware/multipart.js:86:15
[Thu Mar 12 2015 14:52:09 GMT+1100 (AEDT)] ERROR Unable to load /Users/victorhooi/Dropbox/mongodb/glados/scripts/groups: Error: Cannot find module 'underscore'
  at Function.Module._resolveFilename (module.js:336:15)
  at Function.Module._load (module.js:278:25)
  at Module.require (module.js:365:17)
  at require (module.js:384:17)
  at Object.<anonymous> (/Users/victorhooi/Dropbox/mongodb/glados/scripts/groups.coffee:10:7, <js>:4:7)
  at Object.<anonymous> (/Users/victorhooi/Dropbox/mongodb/glados/scripts/groups.coffee:10:1, <js>:67:4)
  at Module._compile (module.js:460:26)
  at Object.loadFile (/Users/victorhooi/Dropbox/mongodb/glados/node_modules/hubot/node_modules/coffee-script/lib/coffee-script/coffee-script.js:182:19)
  at Module.load (/Users/victorhooi/Dropbox/mongodb/glados/node_modules/hubot/node_modules/coffee-script/lib/coffee-script/coffee-script.js:211:36)
  at Function.Module._load (module.js:310:12)
  at Module.require (module.js:365:17)
  at require (module.js:384:17)
  at Robot.loadFile (/Users/victorhooi/Dropbox/mongodb/glados/node_modules/hubot/src/robot.coffee:217:9, <js>:164:20)
  at Robot.load (/Users/victorhooi/Dropbox/mongodb/glados/node_modules/hubot/src/robot.coffee:239:9, <js>:187:30)
  at Shell.loadScripts (/Users/victorhooi/Dropbox/mongodb/glados/node_modules/hubot/bin/hubot:93:5, <js>:94:13)
  at Shell.emit (events.js:104:17)
  at Object.Readline.createInterface.next (/Users/victorhooi/Dropbox/mongodb/glados/node_modules/hubot/src/adapters/shell.coffee:59:9, <js>:104:17)
  at /Users/victorhooi/Dropbox/mongodb/glados/node_modules/hubot/node_modules/readline-history/readline-history.js:112:21
  at ReadStream.onend (/Users/victorhooi/Dropbox/mongodb/glados/node_modules/hubot/node_modules/readline-history/readline-history.js:52:21)
  at ReadStream.emit (events.js:129:20)
  at _stream_readable.js:908:16
  at process._tickCallback (node.js:355:11)
```

It seems in this case, `underscore` wasn't installed as part of the `npm install`.

Is there another step that we need to do as well?

**Comment by victorhooi** [comment]
I should mention that I added the following to my `package.json` in my Hubot directory:
```
    "underscore": "1.8.2"
```
After re-running `npm install`, I didn't get the missing package error messages.

Is this something that should be mentioned as part of the README? `underscore` doesn't seem to come installed by default with Hubot?

**Comment by cycomachead** [comment]
I can't see in that trace that the error is -- it looks like it might be caused from a different package.

`npm install` should automatically put underscore in the proper directory.

I just tried updating the dependcies and pushed a new version, so try installing again or using npm update.

**Comment by cycomachead** [comment]
I think I just got bitten by a similar issue -- this looks like it's because of a longstanding feature/bug in npm about trying to handle duplicate dependencies. (See https://github.com/npm/npm/issues/1341)

This has been fixed in npm 3, AFAIK.

The other solution I had was deleting the module from node_modules and making sure nothing higher-up in the install chain has underscore a depedency, which shouldn't really be more than 1 level of checking.
